src/scripts/install_machine.py

```diff
@@ -0,0 +1,324 @@
+def prepare_cloud_init(args):
+    params = []
+    if args['sourceType'] == 'cloud':
+        params.append("--cloud-init")
+
+        content = "#cloud-config\n"
+        if 'userPassword' in args and 'userLogin' in args:
+            content += "users:\n"
+            content += f"  - name: {args['userLogin']}:{args['userPassword']}\n"
+
+        if 'rootPassword' in args:
+            content += "chpasswd:\n"
+            content += "  list: |\n"
+            content += f"    root:{args['rootPassword']}\n"
+            content += "  expire: False\n"
+
+            user_data_file = tempfile.NamedTemporaryFile(
+                prefix="cockpit-machines-",
+                suffix=("-user-data"),
+                mode='w+'
+            )
+            user_data_file.write(content)
+            user_data_file.flush()
+            params.append(f"user-data={user_data_file.name}")
+
+    yield params
+
+
+def prepare_installation_source(params, args):
+    only_define = False
+    if args['type'] == 'create':
+        if not args['startVm']:
+            only_define = True
+            params.append("--print-xml")
+
+    if args['sourceType'] == "pxe":
+        params += ['--pxe', '--network', args['source']]
+    elif args['sourceType'] == "os":
+        params += ['--install', f"os={args['os']}"]
+    elif args['sourceType'] in ['disk_image', 'cloud']:
+        params.append("--import")
+    elif ((args['source'][0] == '/' and os.path.isfile(args['source'])) or
+            (args['sourceType'] == 'url' and args['source'].endswith(".iso"))):
+        if only_define:
+            return
+        params += ['--cdrom', args['source']]
+    else:
+        if only_define:
+            return
+        params += ['--location', args['source']]
+
+
+@contextmanager
+def prepare_virt_install_params(args):
+    logging.debug(args)
+    with prepare_unattended(args) as unattended_params, prepare_cloud_init(args) as cloud_init_params:
+        params = [
+            "virt-install",
+            "--connect", f"qemu:///{args['connectionName']}",
+            "--quiet",
+            "--name", args['vmName'],
+            "--os-variant", args['os'],
+            "--memory", str(args['memorySize']),
+        ]
+
+        if args['type'] == 'install' or ('storagePool' in args and args['storagePool'] != 'NewVolume'):
+            params += ["--check", "path_in_use=off"]
+
+        if args['sourceType'] != 'disk_image':
+            params += ["--wait", "-1"]
+
+        if args['type'] == 'install' or args['startVm']:
+            params.append("--noautoconsole")
+
+        # Disks
+        if args['type'] == 'install':
+            disks = args['disks'] or ['none']
+            for disk in disks:
+                params += ['--disk', disk]
+        else:
+            params.append("--disk")
+            if args['sourceType'] == "cloud":
+                backing_store = f",backing_store={args['source']},"
+            else:
+                backing_store = ""
```

Comment:
It feels weird that this gets set when it might not be used, depending on which case is picked below.  Maybe delete it from here:

```suggestion
```

and later...